### PR TITLE
[#48243] Convert `project.js` to Stimulus

### PR DESCRIPTION
**Pull request by aaron-contreras**
https://community.openproject.org/work_packages/48243

### CODE REVIEW

File: frontend/src/stimulus/controllers/dynamic/project.controller.ts [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+    if (descriptionRow.classList.contains('-expanded')) {
+      descriptionRow.setAttribute('aria-live', 'polite');
+    } else {
+      descriptionRow.removeAttribute('aria-live');
+    }
+  }
+
+  sendForm() {
+    const ajaxIndicator = document.querySelector('#ajax-indicator') as HTMLElement;
+    ajaxIndicator.style.display = '';
```
Comment (aaron-contreras) [inline frontend/src/stimulus/controllers/dynamic/project.controller.ts:213]: @oliverguenther I tried making this a target but it'd need to be a DOM descendant of the `ProjectController` element.  I tried making the `ajaxIndicator` a controller of its own and using an outlet to be able to reference it but I got:
```
setup.ts:19 Error: Missing outlet element "ajaxIndicator" for "project" controller
```

I made sure to follow the stimulus guides and add a class selector to the outlet element and upon looking up the above error, I got answers relating to outlets also needing to be within the host controller's DOM descendants and to use event emitting instead but I think this is what outlets are trying to solve so I might just have done it wrong. I'd appreciate help with this unless it's overkill for just showing the `ajaxIndicator` and using a simple querySelector would be fine.

Comment (oliverguenther) [reply frontend/src/stimulus/controllers/dynamic/project.controller.ts:213]: Hi @aaron-contreras yes I think that's the right call. We might in the long run have that in its own controller (or replace it with the loading indicator anyway) so being pragmatic here is fine.

File: frontend/src/app/core/augmenting/openproject-augmenting.module.ts [hunk 2]
```diff
@@ -29,17 +29,13 @@
...
 import { OpModalWrapperAugmentService } from 'core-app/shared/components/modal/modal-wrapper-augment.service';
-import { PathScriptAugmentService } from 'core-app/core/augmenting/services/path-script.augment.service';
 
 @NgModule({
   imports: [OpenprojectModalModule],
-  providers: [PathScriptAugmentService],
 })
 export class OpenprojectAugmentingModule {
-  constructor(modalWrapper:OpModalWrapperAugmentService,
-    pathScript:PathScriptAugmentService) {
```
Comment (oliverguenther) [inline frontend/src/app/core/augmenting/openproject-augmenting.module.ts:40]: 👏

**Review by oliverguenther — APPROVED** [review]

**Comment by oliverguenther** [comment]
Nice work @aaron-contreras , couldn't find any issues in my local testing

**Pull request merged**